galasa-ui/src/app/myprofile/page.tsx

```diff
@@ -9,12 +9,24 @@ import { useEffect, useState } from "react";
 import { Loading, ToastNotification } from "@carbon/react";
 import styles from "../../styles/MyProfile.module.css";
 import PageTile from "@/components/PageTile";
+import { FrontEndClient } from "@/utils/interfaces/FrontEndClient";
 
 export default function MyProfilePage() {
 
   const [isLoading, setIsLoading] = useState(false);
   const [isError, setIsError] = useState(false);
-  const [loginId, setLoginId] = useState("");
+
+  const [userData, setUserData] = useState<{
+    loginId: string | null;
+    webLastLogin: string | null;
+    restApiLastLogin: string | null;
```

Comment:
Could this be an interface?

Instead of `string | null`, you can use `field?: string` to indicate that the field hasn't been initialised (i.e. is `string | undefined`).